src/core/dataset.ts

```diff
@@ -26,7 +29,19 @@ module Plottable {
       this._data = data;
       this._metadata = metadata;
       this._accessor2cachedExtent = new Utils.StrictEqualityAssociativeArray();
-      this.broadcaster = new Core.Broadcaster(this);
+      this._callbacks = new Utils.CallbackSet<DatasetCallback>();
+    }
+
+    public onUpdate(callback: DatasetCallback) {
+      this._callbacks.add(callback);
+    }
+
+    public offUpdate(callback: DatasetCallback) {
+      this._callbacks.delete(callback);
+    }
+
+    private _dispatchChange() {
+      this._callbacks.callCallbacks(this);
```

Comment:
This is a one-line private function thats only used twice. Should we maybe jus do `this._callbacks.callCallbacks(this)` whenever we want to dispatch a change?